platform/ui/src/contextProviders/ViewportGridProvider.tsx

```diff
@@ -89,35 +83,44 @@ export function ViewportGridProvider({ children, service }) {
         } = action.payload;
 
         // If empty viewportOptions, we use numRow and numCols to calculate number of viewports
-        const numPanes = layoutOptions.length || numRows * numCols;
-        const viewports = state.viewports.slice();
-        const activeViewportIndex =
-          state.activeViewportIndex >= numPanes ? 0 : state.activeViewportIndex;
-
-        while (viewports.length < numPanes) {
-          viewports.push({});
-        }
-        while (viewports.length > numPanes) {
-          viewports.pop();
-        }
-
-        for (let i = 0; i < numPanes; i++) {
-          let xPos, yPos, w, h;
-
-          if (layoutOptions && layoutOptions[i]) {
-            ({ x: xPos, y: yPos, width: w, height: h } = layoutOptions[i]);
-          } else {
-            const { row, col } = unravelIndex(i, numRows, numCols);
-            w = 1 / numCols;
-            h = 1 / numRows;
-            xPos = col * w;
-            yPos = row * h;
+        const hasOptions = layoutOptions?.length;
+        const viewports = [];
+        let activeViewportIndex;
+
+        for (let col = 0; col < numCols; col++) {
+          for (let row = 0; row < numRows; row++) {
+            const pos = col + row * numCols;
+            const layoutOption = layoutOptions[pos];
+            const viewportId = layoutOption?.viewportId || `${col},${row}`;
```

Comment:
I think `viewport-1-2` is more readable than `1,2`

Reply:
Changed.